assets/CAIP-25/openrpc.json

```diff
@@ -0,0 +1,291 @@
+{
+  "openrpc": "1.2.4",
+  "info": {
+    "title": "CAIP-25",
+    "description": "[CAIP-25](https://github.com/ChainAgnostic/CAIPs/blob/master/CAIPs/caip-25.md) defines an authorization procedure for a chain agnostic provider to interface with a wallet as part of their initialization and/or 'handshake' protocol.",
+    "version": "1.0.0"
+  },
+  "methods": [
+    {
+      "name": "provider_authorization",
+      "paramStructure": "by-position",
```

Comment:
shouldn't this be "by-name"?

Reply:
ah yes, I think we were playing around with this setting when we were chatting.

Reply:
In JSON-RPC, the `params` key can either be an object or an array. The example in CAIP-25 uses the by-name format, where params is an object. It is not specified if this is a requirement or if using params: [] as an array is also allowed.

Reply:
my question was about a request with no `optionalScopes` -- am I misreading or would a response without an optional object still be valid under `by-name` parsing?

Reply:
by-name would be

```
{
  "id": 1,
  "jsonrpc": "2.0",
  "method": "provider_authorization",
  "params": {
    "requiredNamespaces": {
      "eip155": {
        "chains": [
          "eip155:1",
          "eip155:137"
        ],
        "methods": [
          "eth_sendTransaction",
          "eth_signTransaction",
          "eth_sign",
          "get_balance",
          "personal_sign"
        ],
        "events": [
          "accountsChanged",
          "chainChanged"
        ]
      },
      "eip155:10": {
        "methods": [
          "get_balance"
        ],
        "events": [
          "accountsChanged",
          "chainChanged"
        ]
      },
      "cosmos": {}
    },
    "optionalNamespaces": {
      "eip155:42161": {
        "methods": [
          "eth_sendTransaction",
          "eth_signTransaction",
          "get_balance",
          "personal_sign"
        ],
        "events": [
          "accountsChanged",
          "chainChanged"
        ]
      }
    },
    "sessionProperties": {
      "expiry": "2022-12-24T17:07:31+00:00"
    }
  }
}
```

by position would look like:


```
{
    "jsonrpc": "2.0",
    "method": "provider_authorization",
    "params": [
        {
            "eip155": {
                "chains": [
                    "eip155:1",
                    "eip155:137"
                ],
                "methods": [
                    "eth_sendTransaction",
                    "eth_signTransaction",
                    "eth_sign",
                    "get_balance",
                    "personal_sign"
                ],
                "events": [
                    "accountsChanged",
                    "chainChanged"
                ]
            },
            "eip155:10": {
                "methods": [
                    "get_balance"
                ],
                "events": [
                    "accountsChanged",
                    "chainChanged"
                ]
            },
            "cosmos": {}
        },
        {
            "eip155:42161": {
                "methods": [
                    "eth_sendTransaction",
                    "eth_signTransaction",
                    "get_balance",
                    "personal_sign"
                ],
                "events": [
                    "accountsChanged",
                    "chainChanged"
                ]
            }
        },
        {
            "expiry": "2022-12-24T17:07:31+00:00"
        }
    ],
    "id": 0
}
```